### PR TITLE
Ticket - 3 "GET /api/topics

### PR DESCRIPTION
**Pull request by BillyZen**
/api/topics endpoint created with 404 errors including their tests

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+});
+
+describe("/api/topics", () => {
+	describe("GET", () => {
+		test("status: 200 returns an array of topic objects, which have the properties slug and description.", () => {
+		return request(app)
+            .get("/api/topics")
+            .expect(200)
+            .then((response) => {
+                response.body.topics.forEach(topic => {
```
Comment (daviidbartlett) [inline __tests__/app.test.js:20]: please ensure you are actually getting back some topics! if you got back an empty array you wouldn't be running any of these assertions and hence would pass the test

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+const app = require('../app');
+const db = require('../db/connection');
+const seed = require('../db//seeds/seed');
+const testData = require('../db/data/test-data')
+
+beforeEach(() => seed(testData))
+
+afterAll(() => {
+  if (db.end) db.end();
+});
```
Comment (daviidbartlett) [inline __tests__/app.test.js]: why the conditional?

File: app.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+const express = require('express');
+const { getTopics } = require('./Controllers/controllers');
+const dotenv = require('dotenv')
+const app = express();
+dotenv.config();
+
```
Comment (daviidbartlett) [inline app.js:4]: do you need dotenv here?

**Review by daviidbartlett — CHANGES_REQUESTED** [review]
please see comments 😄

File: __tests__/app.test.js (hunk 2 above)
Comment (BillyZen) [reply __tests__/app.test.js]: Think I saw it done that way in a lecture once. Maybe Dave R's? Happy to remove

File: app.js (hunk 3 above)
Comment (BillyZen) [reply app.js:4]: Nope

File: __tests__/app.test.js (hunk 1 above)
Comment (BillyZen) [reply __tests__/app.test.js:20]: added length criteria to test and removed unused code

Comment (daviidbartlett) [reply __tests__/app.test.js:20]: you could probs assert that your get 3 items back?

**Pull request merged**